common/config/src/props.ts

```diff
@@ -92,19 +92,19 @@ export function useProps<T extends Record<string, any>>(
     const getDefault = () =>
       (!isFunc && isFunction(defaultValue) ? defaultValue() : defaultValue) as T[keyof T]
 
-    validator &&
-      watch(
-        () => sourceProps[key],
-        value => {
-          const result = validator(value)
-
-          if (result === false) {
-            console.warn(
-              `${toWarnPrefix(name)}: an invaild value is set to '${key as string}' prop`
-            )
-          }
-        }
-      )
+    const execValidator = (value: T[keyof T]) => {
+      if (!validator || isNull(value)) return
+
+      const result = validator(value)
+
+      if (result === false) {
+        console.warn(`${toWarnPrefix(name)}: an invaild value is set to '${key as string}' prop`)
+      }
+    }
+
+    execValidator(sourceProps[key])
+
+    validator && watch(() => sourceProps[key], execValidator)
```

Comment:
I think it should use the original way to define, and add `{ immediate: true }` as the third parameters of `watch`.